### PR TITLE
Blog app forms

### PR DESCRIPTION
**Pull request by Isabellanamutenda**
In this pull request I:

Create forms to perform these functions:

- Create a post on behalf of the current_user
- Create a comment on behalf of the current_user

Allow users to add likes to posts

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,39 +1,46 @@
...
 
-  def new
-    @post = Post.new
+    @post = Post.find_by(id: params['id'])
+    if @post.user_id == params['user_id'].to_i
+      render '../views/layouts/partials/_userpostdetails'
+    else
+      render '../views/layouts/partials/_wrongid'
+    end
   end
```
Comment (Emmyn5600) [inline app/controllers/posts_controller.rb]: - [x] This rendering way is not appropriate inside the controller. how about doing this:

def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
      
Kindly, refer to the above implementation to work on your controllers.

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1,6 +1,5 @@
 <%  @users.each  do |user| %>
-<a href="/users/<%= user.id %>" class="link" >
-<%= render partial: "user_card", object: user, as: 'user' %>
-</a>
-<% end %>
-
+    <a href="/users/<%= user.id %>" class="link" >
+    <%= render partial: "user_card", object: user, as: 'user' %>
+    </a>
+    <% end %>
```
Comment (Emmyn5600) [inline app/views/users/index.html.erb]: - [x] Kindly, fix this issue that is happening inside the ` app/views/users/index.html.erb`. before you request another review.

blogform

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @Isabellanamutenda 

Great work on making the project requirements 👏 
You've done well implementing some of the project requirements, but there are still some which aren't addressed yet, please work on it and ask for another review.

 ## Required Changes ♻️

- _Check the comment under the review._

## Optional suggestions

- n/a

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/views/users/index.html.erb (hunk 2 above)
Comment (Isabellanamutenda) [reply app/views/users/index.html.erb]: Fixed

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Isabellanamutenda) [reply app/controllers/posts_controller.rb]: Fixed

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**